mirgecom/initializers.py

```diff
@@ -850,3 +851,116 @@ def __call__(self, x_vec, eos, **kwargs):
 
         return make_conserved(dim=self._dim, mass=mass, energy=energy,
                               momentum=mom, species_mass=specmass)
+
+
+class PlanarPoiseuille:
+    r"""Initializer for the planar Poiseuille case.
+
+    The planar Poiseuille case is defined as a viscous flow between two
+    stationary parallel sides with a uniform pressure drop prescribed
+    as *p_hi* at the inlet and *p_low* at the outlet. See the figure below:
+
+    .. figure:: ../figures/poiseuille.png
+        :scale: 50 %
+        :alt: Poiseuille domain illustration
+
+        Illustration of the Poiseuille case setup
+
+    The exact Poiseuille solution is defined by the following:
+    $$
+    P(x) &= P_{\text{hi}} + P'x\\
+    v_x &= \frac{-P'}{2\mu}y(h-y), v_y = 0\\
+    \rho &= \rho_0\\
+    \rho{E} &= \frac{P(x)}{(\gamma-1)} + \frac{\rho}{2}(\mathbf{v}\cdot\mathbf{v})
+    $$
+
+    Here, $P'$ is the constant slope of the linear pressure gradient from the inlet
+    to the outlet and is calculated as:
+    $$
+    P' = \frac{(P_{\text{hi}}-P_{\text{low}})}{\text{length}}
+    $$
+    $v_x$, and $v_y$ are respectively the x and y components of the velocity,
+    $\mathbf{v}$, and $\rho_0$ is the supplied constant density of the fluid.
+    """
+
+    def __init__(self, p_hi=100100., p_low=100000., mu=1.0, height=.02, length=.1,
+                 density=1.0):
+        """Initialize the Poiseuille solution initializer.
+
+        Parameters
+        ----------
+        p_hi: float
+            Pressure at the inlet (default=100100)
+        p_low: float
+            Pressure at the outlet (default=100000)
+        mu: float
+            Fluid viscosity, (default = 1.0)
+        height: float
+            Height of the domain, (default = .02)
+        length: float
+            Length of the domain, (default = .1)
+        density: float
+            Constant density of the fluid, (default=1.0)
+        """
+        self.length = length
+        self.height = height
+        self.dpdx = (p_hi - p_low)/length
```

Comment:
Reverse `hi` and `low`, I think? (And adjust the expressions in the other methods accordingly.)

Reply:
Well, the way I have initializer, the self.dpdx is just the magnitude. The sign is accounted for in `P(x) = P_hi - self.dpdx*x` expression.

Reply:
(4fb140c) 🤞